app/controllers/application_controller.rb

```diff
@@ -2,4 +2,9 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
+  def current_user
+    @_current_user ||= User.find session[:user_id] if session[:user_id]
```

Comment:
Any thoughts on returning a 'Guest' object rather than `nil` if there's no user logged in?